spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala

```diff
@@ -326,11 +326,13 @@ trait CommonJobExecution extends ProjectMetadata {
   }
 
   protected def addInfoColumns(intoDf: DataFrame, reportDate: String, reportVersion: Int): DataFrame = {
-    import za.co.absa.enceladus.utils.implicits.DataFrameImplicits.DataFrameEnhancements
+    import za.co.absa.spark.commons.implicits.DataFrameImplicits.DataFrameEnhancements
+    val function: (DataFrame, String) => DataFrame = (df: DataFrame, _) =>
+      df.withColumn("errCol", lit(Array.emptyIntArray))
     intoDf
-      .withColumnIfDoesNotExist(InfoDateColumn, to_date(lit(reportDate), ReportDateFormat))
-      .withColumnIfDoesNotExist(InfoDateColumnString, lit(reportDate))
-      .withColumnIfDoesNotExist(InfoVersionColumn, lit(reportVersion))
+      .withColumnIfDoesNotExist(function)(InfoDateColumn, to_date(lit(reportDate), ReportDateFormat))
```

Comment:
It is really late, but I think I take issue with `DataFrameImplicits.DataFrameEnhancements.withColumnIfDoesNotExist(ifExists: (DataFrame, String) => DataFrame)(colName: String, colExpr: Column)` that originated in https://github.com/AbsaOSS/spark-commons/pull/18, apparently.

IMHO the name of the method simply does not conform to what it does and what params it requires - the name suggests no-op column does not exist, but it allows an action in such a case, too. My vote is either to have two separate methods `withColumnIfDoesNotExist` and `transformIfColumnExists` or at least rename this method somehow to make sense.

Here, you can alleviate the problem only by renaming the `function` to something more descriptive like `ifExistsFn` or similar. (this occurs in multiple codebase files)

Reply:
I tend to agree. In minimum, the parameters order seems wrong, and could have a default value in the `ifExists` parameter.